### PR TITLE
Swift: More improvements for the string length conflation query

### PR DESCRIPTION
**Pull request by geoffw0**
More improvements for the string length conflation query:
 - added a test for a potential issue with implementations of `NSString` that has been discussed before, where member functions are defined in a parent class rather than directly in `NSString` itself.  At the moment I believe we can't *fix* this issue because we don't have information about base classes, but at least now it is documented with a test.
 - clean up the QL for `isSink` (less code duplication).

### CODE REVIEW

File: swift/ql/src/queries/Security/CWE-135/StringLengthConflation.ql [hunk 1]
```diff
@@ -41,88 +41,80 @@ class StringLengthConflationConfiguration extends DataFlow::Configuration {
...
-        methodName = "substring(to:)" and
-        paramName = "to"
-        or
-        // `NSMutableString.insert`
-        className = "NSMutableString" and
-        methodName = "insert(_:at:)" and
-        paramName = "at"
-      ) and
-      c.getName() = className and
-      c.getAMember() = f and // TODO: will this even work if its defined in a parent class?
```
Comment (jketema) [inline swift/ql/src/queries/Security/CWE-135/StringLengthConflation.ql:76]: I think I need a voice over for the removal of the comment. My (likely flawed) intuition is that the relevant case here is:
```
let nstr2 = ns.substring(from: s.count - 1) // BAD: String length used in NSString [NOT DETECTED]
```
However, that says "not detected"

**Review by jketema — COMMENTED** [review]
One question. Otherwise this looks very nice.

File: swift/ql/src/queries/Security/CWE-135/StringLengthConflation.ql (hunk 1 above)
Comment (rdmarsh2) [reply swift/ql/src/queries/Security/CWE-135/StringLengthConflation.ql:76]: That's what I'd expect - we typically don't keep TODO comments in the code once there's a failing test case for them.

Comment (jketema) [reply swift/ql/src/queries/Security/CWE-135/StringLengthConflation.ql:76]: Thanks for clearing that up.

**Review by jketema — APPROVED** [review]

File: swift/ql/src/queries/Security/CWE-135/StringLengthConflation.ql (hunk 1 above)
Comment (geoffw0) [reply swift/ql/src/queries/Security/CWE-135/StringLengthConflation.ql:76]: Yep, sorry for not replying sooner.  We generally don't like to keep TODO comments in the code, but sadly this is unfixable at the moment, so its documented with a NOT DETECTED test case now.

Should be an easy fix when we extract information about class hierarchies.

**Pull request merged**